File: api/src/main/java/net/blueshell/api/common/util/MappingUtil.java
package net.blueshell.api.common.util;

import net.blueshell.api.model.User;

import java.security.SecureRandom;
import java.util.function.BiConsumer;

public class MappingUtil {
    private static final String CHAR_SET = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%&*()-_=+<>?";
    private static final int PASSWORD_LENGTH = 12;
    private static final SecureRandom random = new SecureRandom();


    public static <O, T> void applyIfFieldIsNotNull(O user, T obj, BiConsumer<O, T> applier) {
        if (obj != null) {
            applier.accept(user, obj);
        }
    }

    public static String generatePassword() {
        StringBuilder password = new StringBuilder(PASSWORD_LENGTH);
        for (int i = 0; i < PASSWORD_LENGTH; i++) {
            int index = random.nextInt(CHAR_SET.length());
            password.append(CHAR_SET.charAt(index));
        }
        return password.toString();
    }
}

File: api/src/main/kotlin/net/blueshell/api/common/util/MappingUtil.kt
package net.blueshell.api.common.util

import java.security.SecureRandom
import java.util.function.BiConsumer

object MappingUtil {
    private const val CHAR_SET = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%&*()-_=+<>?"
    private const val PASSWORD_LENGTH = 12
    private val random = SecureRandom()


    fun <O, T> applyIfFieldIsNotNull(user: O?, obj: T?, applier: BiConsumer<O?, T?>) {
        if (obj != null) {
            applier.accept(user, obj)
        }
    }

    fun generateRandomString(): String {
        val password = StringBuilder(PASSWORD_LENGTH)
        for (i in 0..<PASSWORD_LENGTH) {
            val index = random.nextInt(CHAR_SET.length)
            password.append(CHAR_SET[index])
        }
        return password.toString()
    }
}
